lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -144,8 +146,17 @@ def index(self, trans, deleted='False', **kwd):
         filters = []
         # support the old default of not-returning/filtering-out deleted histories
         filters += self._get_deleted_filter(deleted, filter_params)
-        # users are limited to requesting only their own histories (here)
-        filters += [self.app.model.History.user == current_user]
+        # check if user is admin and get optional parameter 'all' 
+        admin = trans.user_is_admin
```

Comment:
Minor suggestion. You only need to check the user's admin status if `all_history` is `True`. So you could move this line inside the `if` stmt; but you could also get rid of it completely and do `if not trans.user_is_admin` instead - which, I think, will read even better.

Reply:
Added. Thanks mate

Reply:
looks great!